packages/browser/src/types/analytics.ts

```diff
@@ -26,41 +26,70 @@ export interface ExperimentAnalyticsEvent {
    * User properties to identify with the user prior to sending the event.
    */
   userProperties?: Record<string, unknown>;
-}
-
-/**
- * Event for tracking a user's exposure to a variant. This event will not count
- * towards your analytics event volume.
- */
-export class ExposureEvent implements ExperimentAnalyticsEvent {
-  name = '[Experiment] Exposure';
-  properties: Record<string, string>;
-  userProperties?: Record<string, unknown>;
 
   /**
    * The user exposed to the flag/experiment variant.
    */
-  public user: ExperimentUser;
+  user: ExperimentUser;
 
   /**
    * The key of the flag/experiment that the user has been exposed to.
    */
-  public key: string;
+  key: string;
 
   /**
    * The variant of the flag/experiment that the user has been exposed to.
    */
-  public variant: Variant;
+  variant: Variant;
+
+  /**
+   * The user property for the flag/experiment (auto-generated from the key)
+   */
+  userProperty: string;
+}
 
-  public constructor(user: ExperimentUser, key: string, variant: Variant) {
-    this.key = key;
-    this.variant = variant;
-    this.properties = {
-      key: key,
-      variant: variant.value,
-    };
-    this.userProperties = {
-      [`[Experiment] ${key}`]: variant.value,
-    };
-  }
+/**
+ * Event for tracking a user's exposure to a variant. This event will not count
+ * towards your analytics event volume.
+ */
+export const exposureEvent = (
+  user: ExperimentUser,
+  key: string,
+  variant: Variant,
+  source: VariantSource,
+): ExperimentAnalyticsEvent => {
+  const name = '[Experiment] Exposure';
+  const value = variant?.value;
+  const userProperty = `[Experiment] ${key}`;
+  return {
+    name,
+    user,
+    key,
+    variant,
+    userProperty,
+    properties: {
+      key,
+      variant: value,
+      source,
+    },
+    userProperties: {
+      [userProperty]: value,
+    },
+  };
+};
+
+export enum VariantSource {
+  LOCAL_STORAGE = 'storage',
+  INITIAL_VARIANTS = 'initial',
+  SECONDARY_LOCAL_STORAGE = 'secondary-storage',
+  SECONDARY_INITIAL_VARIANTS = 'secondary-initial',
+  FALLBACK_INLINE = 'fallback-inline',
+  FALLBACK_CONFIG = 'fallback-config',
 }
+
+export const isFallback = (source: VariantSource): boolean => {
+  return (
+    source === VariantSource.FALLBACK_INLINE ||
+    source === VariantSource.FALLBACK_CONFIG
+  );
+};
```

Comment:
Not sure this belongs in this file.

Could put this and `Source` from config in its own file within `/types`

Reply:
👍